monai/apps/deepgrow/handler.py

```diff
@@ -0,0 +1,287 @@
+        for region in self.metric_data:
+            metric = self.metric_data[region].mean()
+            self.logger.info(
+                "Epoch[{}] Metrics -- Region: {:0>2d}, {}: {:.4f}".format(epoch, region, self.tag_name, metric)
+            )
+
+            if self.merge_scalar:
+                means["{:0>2d}".format(region)] = metric
+            else:
+                self.writer.add_scalar("{}_{:0>2d}".format(self.tag_name, region), metric, epoch)
+            metric_sum += metric
+
+        if self.merge_scalar:
+            means["avg"] = metric_sum / len(self.metric_data)
+            self.writer.add_scalars("{}_region".format(self.tag_name), means, epoch)
+        elif len(self.metric_data) > 1:
+            metric_avg = metric_sum / len(self.metric_data)
+            self.writer.add_scalar("{}_regions_avg".format(self.tag_name), metric_avg, epoch)
+        self.writer.flush()
+
+    def __call__(self, engine: Engine, action) -> None:
+        total_steps = engine.state.iteration
+        if total_steps < engine.state.epoch_length:
+            total_steps = engine.state.epoch_length * (engine.state.epoch - 1) + total_steps
+
+        if action == "epoch" and not self.fold_size:
+            epoch = engine.state.epoch
+        elif self.fold_size and total_steps % self.fold_size == 0:
+            epoch = int(total_steps / self.fold_size)
+        else:
+            epoch = None
+
+        if epoch:
+            if self.images and epoch % self.image_interval == 0:
+                self.write_images(epoch)
+            if self.add_scalar:
+                self.write_region_metrics(epoch)
+
+        if action == "epoch" or epoch:
+            self.plot_data = {}
+            self.metric_data = {}
+            return
+
+        device = engine.state.device
+        batch_data = engine.state.batch
+        output_data = engine.state.output
+
+        for bidx in range(len(batch_data.get("region", []))):
+            region = batch_data.get("region")[bidx]
+            region = region.item() if torch.is_tensor(region) else region
+
+            if self.images and self.plot_data.get(region) is None:
+                self.plot_data[region] = [
+                    rescale_array(batch_data["image"][bidx][0].detach().cpu().numpy()[np.newaxis], 0, 1),
+                    rescale_array(batch_data["label"][bidx].detach().cpu().numpy(), 0, 1),
+                    rescale_array(output_data["pred"][bidx].detach().cpu().numpy(), 0, 1),
+                ]
+
+            if self.compute_metric:
+                if self.metric_data.get(region) is None:
+                    self.metric_data[region] = MeanDice()
+                self.metric_data[region].update(
+                    y_pred=output_data["pred"][bidx].to(device), y=batch_data["label"][bidx].to(device), batched=False
+                )
+
+
+class SegmentationSaver:
+    def __init__(
+        self,
+        output_dir: str = "./runs",
+        save_np=False,
+        images=True,
```

Comment:
if "images" is False then this class is not doing anything, so I think we can remove this arg?

Reply:
removed this code.. not need as part of core functionality for deepgrow